### ISSUE TITLE
Feature Request: AI-Enhanced Rebases

### ISSUE THREAD

**Issue by ctoth** [paul-gauthier/aider#360]
## Motivation
I have been learning that it's best to do changes in very small chunks, (For instance update an interface, then modify its implementations one-by-one.
This leaves me with tons and tons of tiny commit messages, each with a bunch of useful information.

## Solution
What would be incredible is if we could issue a /squash command, where Aider passes these commits (and diffs of source code?) and a prompt off requesting they all be summarized and combined so that once a feature is complete, it has a clean and wonderful single commit.

There are existing tools that do GPT-powered commit messages, so we could probably take some inspiration from there.

I'd be happy to put in a PR if this is a feature you would be interested in, though I'd like to discuss how it should fit into the existing architecture.

**Comment by ctoth** [comment]
I see we already have /commit which does the auto-commit work, so /squash should be even simpler :)

**Comment by paul-gauthier** [comment]
I think this is out of scope for aider at this time.
